test/tests/functional/pbs_qsub_direct_write.py

```diff
@@ -106,3 +226,49 @@ def test_qalter_direct_write(self):
                                    id=jid)
         except PbsAlterError as e:
             print str(e)
+
+    def test_qalter_direct_write_error(self):
+        """
+        submit a job and after it starts running alter
+        the job with -koed and check whether expected
+        error message appears
+        """
+        j = Job(TEST_USER)
+        jid = self.server.submit(j)
+        attribs = {ATTR_k: 'oed'}
+        self.server.expect(JOB, {'job_state': 'R'})
+        try:
+            self.server.alterjob(jid, attribs)
+        except PbsAlterError as e:
+            self.assertTrue(
+                'qalter: Cannot modify attribute while job running  Keep_Files'
+                in e.msg[0])
+
+    def test_direct_write_qrerun(self):
+        """
+        submit a sleep job and make sure that the std_files
+        are written and when a job is rerun error message
+        in logged in mom_log that it is skipping directly
+        written/absent spool file as files are already
+        present on first run of the job.
+        """
+        self.mom.add_config({'$logevent': '0xffffffff'})
+        j = Job(TEST_USER)
+        j.set_attributes({ATTR_k: 'doe'})
```

Comment:
Same as above. You do not need a separate `set_attributes` call. Please also fix other places.